### DISCUSSION TITLE
PyCaret AUC is overestimating the actual performance.

### DISCUSSION THREAD

**Discussion by sorenwacker** [pycaret/pycaret#3355] [General]
![image](https://user-images.githubusercontent.com/3391614/222558389-df1a2bcb-0f18-4bdd-bbec-e14d20ab5201.png)

I wonder if the implementation of pycaret AUC values should be changed. The AUC values that e.g. `create_model` reports are not real AUC values, but somehow weighted. So, they would be very different from conventional AUC values calculated from ROC curves with e.g. scikit-learn. Unfortunately, pycaret does not hint the user to this issue e.g. by calling the metric wAUC for example (weighted AUC).

What makes things even worse is that the values seem to not reflect the actual model performance well. In the picture above you can see how the AUC values relate to other metrics. Even a rather bad model F1 ~0.175 gives quite acceptible AUC values in the pycaret framework AUC ~0.73.

The AUC is already intrisically weighting according to class-imbalances, so why not use the conventional AUC?

**Comment by moezali1** [comment]
Is it for multiclass or binary classification?

**Comment by sorenwacker** [comment]
This is binary.
________________________________

**Comment by Yard1** [comment]
Hi @sorenwacker, we are using the sklearn implementation of ROC-AUC. While we do use weighted ROC-AUC for multiclass problems, the arguments we set should be ignored for the binary case according to sklearn documentation.

**Comment by sorenwacker** [comment]
I will check that, maybe I am doing something wrong.
